modules/systemdunits/systemdunits.go

```diff
@@ -0,0 +1,86 @@
+package systemdunits
+
+import (
+	"github.com/coreos/go-systemd/v22/dbus"
+	"github.com/netdata/go-orchestrator/module"
+	"github.com/netdata/go.d.plugin/pkg/matcher"
+)
+
+type Config struct {
+	Selector matcher.SimpleExpr `yaml:"selector"`
+}
+
+func init() {
+	creator := module.Creator{
+		Defaults: module.Defaults{
+			Disabled:    false,
+			UpdateEvery: 1,
+		},
+		Create: func() module.Module { return New() },
+	}
+
+	module.Register("systemdunits", creator)
+}
+
+// New creates SystemdUnits with default values
+func New() *SystemdUnits {
+	return &SystemdUnits{
+		collectedUnits: make(map[string]bool),
+		charts:         charts.Copy(),
+	}
+}
+
+// SystemdUnits systemdunits module
+type SystemdUnits struct {
+	module.Base    // should be embedded by every module
+	Config         `yaml:",inline"`
+	charts         *module.Charts
+	collectedUnits map[string]bool
+	units          []dbus.UnitStatus
+	selector       matcher.Matcher
+}
+
+// Cleanup makes cleanup
+func (SystemdUnits) Cleanup() {}
+
+// Init makes initialization
+func (s *SystemdUnits) Init() bool {
+	if !s.Selector.Empty() {
+		m, err := s.Selector.Parse()
+		if err != nil {
+			s.Errorf("error on creating per user stats matcher : %v", err)
+		}
+		s.selector = matcher.WithCache(m)
+	}
+
+	allUnits, err := s.getUnits()
```

Comment:
why `getUnits()` in Init? We call collector's Init functuin only once. `getUnits` should be in collect

Reply:
that's where I need your advice, I did this because I want to keep `dbus` connections outside `collect` method due to having some fake data while we are running tests. any idea/sample how to handle this?